python/pyarrow/io.pyx

```diff
@@ -340,16 +338,37 @@ cdef class MemoryMappedFile(NativeFile):
     cdef:
         object path
 
-    def __cinit__(self, path, mode='r'):
+    def __cinit__(self):
+        self.is_readable = self.is_writeable = 0
```

Comment:
I would prefer if we refrain from using this syntax. It might save a line but it less readable.